### PR TITLE
SONARJAVA-1451 - Handle simple equality between symbolic values

### PR DESCRIPTION
**Pull request by DidierBesset**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java [hunk 1]
```diff
@@ -418,4 +449,46 @@ public String toString() {
...
+        return ids[0] == key.ids[0] && ids[1] == key.ids[1];
+      }
+      return super.equals(obj);
+    }
+
+    @Override
+    public int hashCode() {
+      long bits = ids[0];
+      bits ^= 31 * ids[1];
+      return (((int) bits) ^ ((int) (bits >> 32)));
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java]: MAJOR Remove those useless parentheses. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by DidierBesset** [comment]
Transitivity may require a strong refactoring of the class AbstractBinaryConstraint and subclasses. So, shall we wait for final commit?

File: java-squid/src/main/java/org/sonar/java/se/AbstractBinaryConstraint.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se;
+
+import org.sonar.java.se.ConstraintManager.BooleanConstraint;
+
+public abstract class AbstractBinaryConstraint {
+
+  public abstract boolean comptatibleWithEqual(BooleanConstraint booleanConstraint);
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/se/AbstractBinaryConstraint.java]: typo

File: java-squid/src/main/java/org/sonar/java/se/AbstractBinaryConstraint.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  public abstract boolean comptatibleWithNotEqual(BooleanConstraint booleanConstraint);
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/se/AbstractBinaryConstraint.java]: typo

**Pull request closed without merging**